Tasks/Common/nuget-task-common/NuGetToolRunner.ts

```diff
@@ -139,7 +141,11 @@ function locateTool(tool: string, opts?: LocateOptions) {
 
 export function locateNuGetExe(userNuGetExePath: string): string {
     if (userNuGetExePath) {
-        tl.debug(`using user-supplied NuGet path ${userNuGetExePath}`)
+        if (os.platform() === "win32") {
```

Comment:
I think this is likely fine in your case.  We discussed for lib and the reason we didn't do it there is double quote is valid in OSX and Linux in a dir name.  So we can't blindly do it in a base lib.